File: investments/src/main/java/com/abm/investments/model/Investment.java
package com.abm.investments.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "investments")
public class Investment {

    @Id
    private String id;
    private String name;
    private Amount amount;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Amount getAmount() {
        return amount;
    }

    public void setAmount(Amount amount) {
        this.amount = amount;
    }

    @Override
    public String toString() {
        return "Investment{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", amount=" + amount +
                '}';
    }
}

File: property-manager/src/main/scala/com/propertymanager/domain/Investment.scala
package com.propertymanager.domain

import com.fasterxml.jackson.annotation.{JsonCreator, JsonProperty}
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "investments")
class Investment @JsonCreator()(@JsonProperty("itsType") val itsType: String, @JsonProperty("amount") val amount: Double) {

  @Id
  private val id: String = null

  override def toString = s"Investment($id, $itsType, $amount)"
}
